irctest/server_tests/messages.py

```diff
@@ -123,26 +125,31 @@ def testNoticeNonexistentChannel(self):
 
 
 class TagsTestCase(cases.BaseServerTestCase):
+    @pytest.mark.parametrize("tag_length", [4096, 10000])
     @cases.mark_capabilities("message-tags")
     @cases.xfailIf(
-        lambda self: bool(
+        lambda self, tag_length: bool(
             self.controller.software_name == "UnrealIRCd"
             and self.controller.software_version == 5
         ),
         "UnrealIRCd <6.0.7 dropped messages with excessively large tags: "
         "https://bugs.unrealircd.org/view.php?id=5947",
     )
-    def testLineTooLong(self):
+    def testLineTooLong(self, tag_length):
         self.connectClient("bar", capabilities=["message-tags"], skip_if_cap_nak=True)
         self.connectClient(
             "recver", capabilities=["message-tags"], skip_if_cap_nak=True
         )
         self.joinChannel(1, "#xyz")
-        monsterMessage = "@+clientOnlyTagExample=" + "a" * 4096 + " PRIVMSG #xyz hi!"
+
+        monsterMessage = (
+            "@+clientOnlyTagExample=" + "a" * tag_length + " PRIVMSG #xyz hi!"
+        )
         self.sendLine(1, monsterMessage)
         self.assertEqual(self.getMessages(2), [], "overflowing message was relayed")
```

Comment:
Technically this check should be done after `self.getMessages(1)` so it doesn't race.

Reply:
https://github.com/progval/irctest/pull/288